.github/workflows/build.yml

```diff
@@ -197,7 +197,7 @@ jobs:
           cache-name: ${{ runner.os }}-cache-tools
         with:
           path: ./sources/tools
-          key: ${{ runner.os }}-tools-${{ hashFiles('./sources/nuttx/tools/ci/cibuild.sh') }}
+          key: ${{ runner.os }}-tools-${{ hashFiles('./sources/nuttx/tools/ci/platforms/darwin.sh') }}
```

Comment:
why not use the same entry point

Reply:
Because if any macOS package is changed in darwin.sh the cache is not updated because the cibuild.sh file is still the same. 
Example in the caches list
https://github.com/apache/nuttx/actions/caches

With hashFiles cibuild.sh 1.8 GB cached
macOS-tools-39f0adfce6ad8961ae10c2290191d77969616f26734fb8d1a8a94081cb2d38d7

With hashFiles darwin.sh 2 GB cached
macOS-tools-b17837aa7ba02ebb1c76d3127835a2c092d001431c2648ab79279217fc45f4dd

Reply:
The cache algo is poor.:(